homeassistant/components/cups/sensor.py

```diff
@@ -49,23 +54,43 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     host = config.get(CONF_HOST)
     port = config.get(CONF_PORT)
     printers = config.get(CONF_PRINTERS)
+    is_cups = config.get(CONF_IS_CUPS_SERVER)
 
-    try:
-        data = CupsData(host, port)
+    if is_cups:
+        data = CupsData(host, port, None)
         data.update()
-    except RuntimeError:
-        _LOGGER.error("Unable to connect to CUPS server: %s:%s", host, port)
-        return False
+        if data.available is False:
+            _LOGGER.error("Unable to connect to CUPS server: %s:%s",
+                          host, port)
+            raise PlatformNotReady()
 
-    dev = []
-    for printer in printers:
-        if printer in data.printers:
-            dev.append(CupsSensor(data, printer))
-        else:
-            _LOGGER.error("Printer is not present: %s", printer)
-            continue
+        dev = []
+        for printer in printers:
+            if printer in data.printers:
+                dev.append(CupsSensor(data, printer))
+            else:
+                _LOGGER.error("Printer is not present: %s", printer)
+                continue
 
-    add_entities(dev, True)
+        add_entities(dev, True)
+
+    else:
```

Comment:
Instead of `else`, add a `return` inside the previous if. That way you can save on indentation .